File: dayEight/Java/QuestionTwo.java
public class QuestionTwo {

    public static boolean isBracketComplete(String brackets){
        int openBracketCounter = 0;
        int closeBracketCounter = 0;
        int openSquareBracketCounter = 0;
        int closeSquareBracketCounter = 0;
        int openCurlyBracesCounter = 0;
        int closeCurlyBracesCounter = 0;
        int openAngleBracketsCounter = 0;
        int closeAngleBracketsCounter = 0;
        boolean isComplete = false;
        
        for(int index = 0; index < brackets.length(); index++){

            
            if(brackets.charAt(index) == '('){
                openBracketCounter++;
            }
            if(brackets.charAt(index) == ')'){
                closeBracketCounter++;
            }
            if(brackets.charAt(index) == '['){
                openSquareBracketCounter++;
            }
            if(brackets.charAt(index) == ']'){
                closeSquareBracketCounter++;
            }
            if(brackets.charAt(index) == '{'){
                openCurlyBracesCounter++;
            }
            if(brackets.charAt(index) == '}'){
                closeCurlyBracesCounter++;
            }
            if(brackets.charAt(index) == '<'){
                openAngleBracketsCounter++;
            }
            if(brackets.charAt(index) == '>'){
                closeAngleBracketsCounter++;
            }
        }
        
        if((openAngleBracketsCounter == closeAngleBracketsCounter) && (openBracketCounter == closeBracketCounter) && (openCurlyBracesCounter == closeCurlyBracesCounter) && (openSquareBracketCounter == closeSquareBracketCounter)){
            isComplete = true;
        }
        
        return isComplete;
    }
}

File: dayEight/python/question_two.py
def isBracketComplete(brackets):
    openBracketCounter = 0;
    closeBracketCounter = 0;
    openSquareBracketCounter = 0;
    closeSquareBracketCounter = 0;
    openCurlyBracesCounter = 0;
    closeCurlyBracesCounter = 0;
    openAngleBracketsCounter = 0;
    closeAngleBracketsCounter = 0;
    isComplete = false;
    
    for index in range(len(brackets)):

        
        if(brackets[index] == "("):
            openBracketCounter += 1
        
        if(brackets[index] == ")"):
            closeBracketCounter += 1
        
        if(brackets[index] == "["):
            openSquareBracketCounter += 1
        
        if(brackets[index] == "]"):
            closeSquareBracketCounter += 1
        
        if(brackets[index] == "{"):
            openCurlyBracesCounter += 1
        
        if(brackets[index] == "}"):
            closeCurlyBracesCounter += 1
        
        if(brackets[index] == "<"):
            openAngleBracketsCounter += 1
        
        if(brackets[index] == ">"):
            closeAngleBracketsCounter += 1
    
    if((openAngleBracketsCounter == closeAngleBracketsCounter) and (openBracketCounter == closeBracketCounter) and (openCurlyBracesCounter == closeCurlyBracesCounter) and (openSquareBracketCounter == closeSquareBracketCounter)):
        isComplete = true;
    
    
    return isComplete;

